src/ocrd/processor/base.py

```diff
@@ -63,15 +67,15 @@ class Processor():
 
     def __init__(
             self,
-            workspace : Workspace,
+            # FIXME: deprecate in favor of process_workspace(workspace)
+            workspace : Optional[Workspace],
             ocrd_tool=None,
             parameter=None,
-            # TODO OCR-D/core#274
-            # input_file_grp=None,
-            # output_file_grp=None,
-            input_file_grp="INPUT",
-            output_file_grp="OUTPUT",
+            input_file_grp=None,
+            output_file_grp=None,
             page_id=None,
+            download_files=True,
```

Comment:
That flag is appreciated. Even better if the behavior could be also globally overwritten by an environmental variable.

Reply:
Yes, good idea. We could do that in the top CLI decorator.

But then we must have a well defined behaviour for those input files that do not exist locally: skip or abort. At the moment it will crash during `ocrd_modelfactory.page_from_file`.

Reply:
> But then we must have a well defined behaviour for those input files that do not exist locally: skip or abort.

Agreed. Should be aborted by default.

Reply:
> Seconded, we want to fail early if downloading is disabled and file does not exist. Otherwise there would be hard-to-debug incomplete workspaces at the end because some files would be silently skipped.

Both `abort` and `skip` have valid use-cases IMO. The latter could be data distribution across nodes, for example. Also, if some files are missing inadvertently, we might still want to continue with the rest of pages (as we will for page-wise error handling). After all, missing result pages can always be diagnosed directly from the METS, plus one could inspect/filter the logs accordingly. And we should always be able to re-run after remedying affected pages.

> I would add a `ocrd_utils.config` variable `OCRD_PROCESSOR_DOWNLOADS` which, if set, overrides the behavior of the kwarg. And of course add the config to the `ocrd --help` output. 

Yes, but I would call it `OCRD_DOWNLOAD_INPUT[=true]`. And we should also apply `OCRD_DOWNLOAD_RETRIES` and `OCRD_DOWNLOAD_TIMEOUT` here (and change their description – they currently only cover resmgr activity but are documented as covering only workspace files).

For the missing pages (after failed downloads or if downloads are not even attempted), we could then apply `OCRD_MISSING_INPUT[=abort]` or `=skip`.

Regarding runtime error handling, esp. in light of #579, we should _also_ have some flexible configuration: How about `OCRD_MISSING_OUTPUT[=abort]` or `=skip` or `=copy` (with ocrd-dummy behaviour)?

Reply:
More ideas from today's VC discussion:

When fixing the [conflation of skip=first=last](https://github.com/OCR-D/core/commit/64b356384be8155797ae6147f23447773992c2b0) in `Processor.zip_input_files(on_error=...)`, to have a better separate of concerns we should also raise specifc exceptions (say `MissingInputFile`) cases, so that  a processor developer can still decide to have `on_error=skip` (no exceptions) or `on_error=abort` (exceptions possibly still caught by user if `OCRD_MISSING_INPUT=skip`).

We could also add `OCRD_EXISTING_OUTPUT[=abort]` or `=skip` or `=overwrite` (instead of indiscriminate global `--overwrite`), so the user could run with `OCRD_MISSING_INPUT=skip OCRD_MISSING_OUTPUT=skip` and then attempt re-run any number of times with `OCRD_EXISTING_OUTPUT=skip`.

Further, we should make still make sure that failed pages (regardless of OCRD_MISSING_OUTPUT strategy) get logged in a directly re-usable (self-contained) way, for example `self.logger.critical(json.dumps({ 'output_file_grp': self.output_file_grp, 'page_id': page_id, 'strategy': 'copy', 'error': str(e)}))`.